components/brave_new_tab_ui/components/default/page/index.tsx

```diff
@@ -467,13 +467,26 @@ export const IconButtonContainer = styled('div')<IconButtonContainerProps>`
   font-family: ${p => p.theme.fontFamily.heading};
   font-size: 13px;
   font-weight: 600;
-  color: rgba(255,255,255,0.8);
+  color: rgba(var(--override-readability-color-rgb, 255, 255, 255), 0.8);
   margin-right: ${p => p.textDirection === 'ltr' && '8px'};
   margin-left: ${p => p.textDirection === 'rtl' && '8px'};
-  border-right: ${p => p.textDirection === 'ltr' && '1px solid rgba(255, 255, 255, 0.6)'};
-  border-left: ${p => p.textDirection === 'rtl' && '1px solid rgba(255, 255, 255, 0.6)'};
+  border-right: ${p => p.textDirection === 'ltr' && 'rgba(var(--override-readability-color-rgb, 255, 255, 255), 0.6)'};
+  border-left: ${p => p.textDirection === 'rtl' && 'rgba(var(--override-readability-color-rgb, 255, 255, 255), 0.6)'};
 
   &:hover {
-    color: #ffffff;
+    color: ${p => p.color};
+  }
+`
+
+export const OverrideReadabilityColor = createGlobalStyle<{override: boolean}>`
+  body {
+    ${p => p.override && css`
+      --override-readability-color-rgb: 0, 0, 0;
+      --override-readability-color: rgb(0, 0, 0);
+      
+      // override color property in resets.css. Not sure why this happens but,
+      // the value in the stylesheet wins over variables above.
+      color: inherit
```

Comment:
I couldn't figure out why this happens. This happens when we declare variables with setProperty(). I tried setting property for `body`, but it didn't work. As a workaround, I override `color` here.

The previous trial can be found at https://github.com/brave/brave-core/pull/15230/commits/27a0d34e36d85962ef0b7d71d98762f57421151a

<img width="352" alt="Screen Shot 2022-09-28 at 4 59 33 PM" src="https://user-images.githubusercontent.com/5474642/192747424-9cf8e159-428b-4440-8ca3-f6ce547b2592.png">

Reply:
I think I like yours better anyway :smile: It's weird we need to set the `color: inherit` though - I just tested removing it and it seems to work without

```ts
css`
      --override-readability-color-rgb: 0, 0, 0;
      --override-readability-color: rgb(0, 0, 0);
`
```

Reply:
Ah yeah, I found the culprit. I used wrong variable for some UI.

```diff
-  color: var(--override-readability-color-rgb, #FFFFFF);  <--! oops, rgb is 0, 0, 0 -->
+  color: var(--override-readability-color, #FFFFFF);
```